library/src/main/java/com/google/maps/android/clustering/view/DefaultClusterRenderer.java

```diff
@@ -737,34 +737,139 @@ public void remove(Marker m) {
+        }
+    }
+
     /**
      * Called before the marker for a Cluster is added to the map.
      * The default implementation draws a circle with a rough count of the number of items.
+     *
+     * The first time ClusterManager.cluster() is invoked on a set of items
+     * onBeforeClusterRendered() will be called and onClusterUpdated() will not be called.
+     * If an item is removed and re-added (or updated) and ClusterManager.cluster() is invoked
+     * again, then onClusterUpdated() will be called and onBeforeClusterRendered() will not be
+     * called.
+     *
+     * @param cluster cluster to be rendered
+     * @param markerOptions markerOptions representing the provided cluster
      */
     protected void onBeforeClusterRendered(Cluster<T> cluster, MarkerOptions markerOptions) {
+        // TODO: consider adding anchor(.5, .5) (Individual markers will overlap more often)
+        markerOptions.icon(getDescriptorForCluster(cluster));
+    }
+
+    /**
+     * Gets a BitmapDescriptor for the given cluster that contains a rough count of the number of
+     * items. Used to set the cluster marker icon in the default implementations of
+     * onBeforeClusterRendered() and onClusterUpdated().
+     *
+     * @param cluster cluster to get BitmapDescriptor for
+     * @return a BitmapDescriptor for the marker icon for the given cluster that contains a rough
+     * count of the number of items.
+     */
+    protected BitmapDescriptor getDescriptorForCluster(Cluster<T> cluster) {
         int bucket = getBucket(cluster);
         BitmapDescriptor descriptor = mIcons.get(bucket);
         if (descriptor == null) {
             mColoredCircleBackground.getPaint().setColor(getColor(bucket));
             descriptor = BitmapDescriptorFactory.fromBitmap(mIconGenerator.makeIcon(getClusterText(bucket)));
             mIcons.put(bucket, descriptor);
         }
-        // TODO: consider adding anchor(.5, .5) (Individual markers will overlap more often)
-        markerOptions.icon(descriptor);
+        return descriptor;
     }
 
     /**
      * Called after the marker for a Cluster has been added to the map.
+     *
+     * @param cluster the cluster that was just added to the map
+     * @param marker the marker representing the cluster that was just added to the map
      */
     protected void onClusterRendered(Cluster<T> cluster, Marker marker) {
     }
 
+    /**
+     * Called when a cached marker for a Cluster already exists on the map so the marker may
+     * be updated to the latest cluster values. Default implementation updated the icon with a
+     * circle with a rough count of the number of items. Note that the contents of the cluster may
+     * not have changed since the cached marker was created - implementations of this method are
+     * responsible for checking if something changed (if that matters to the implementation).
+     *
+     * The first time ClusterManager.cluster() is invoked on a set of items
```

Comment:
Nit: use Javadoc links `The first time {@link ClusterManager#cluster()}...`

Reply:
Done in https://github.com/googlemaps/android-maps-utils/pull/627/commits/556d20c2f9655569eaa2f972410e96ec6c381619